service/dap/server.go

```diff
@@ -442,10 +447,18 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
-	mode, ok := request.Arguments["mode"]
-	if !ok || mode == "" {
+	// Validate launch request mode
+	modeArg := request.Arguments["mode"]
+	mode, ok := modeArg.(string)
+	if mode == "" {
```

Comment:
The new code doesn't make it obvious that missing value, not just empty value, is also treated as "debug". I guess you are implicitly relying on string conversion failing if the value is not present because modeArg is nil,  aren't you? But won't this happen if mode is some other type as well? Just like with operations that return non-nil err, I don't think you are supposed to rely on the default value of `mode` being "" if string conversion fails. You should be checking ok right away. Doesn't the check later invalidate treating nil as "debug"? I suggest going back to the original code and using interface keys in your map. If you think we should alter the default behavior, let's do that in a separate change against the new tests I added, so it's clear what's going on.

Reply:
Because of the change to the switch instead of the map these additional type checks are no longer necessary so I rolled back this